modules/database/src/ioc/db/dbCaTest.c

```diff
@@ -45,29 +45,106 @@
 #include "dbCa.h"
 #include "dbCaPvt.h"
 #include "dbCaTest.h"
-#include "dbCommon.h"
 #include "db_convert.h"
 #include "dbLock.h"
 #include "link.h"
 
-
+typedef struct ca_counts {
+    int level;
+    int ncalinks;
+    int nconnected;
+    unsigned long nDisconnect;
+    unsigned long nNoWrite;
+    int noReadAccess;
+    int noWriteAccess;
+} ca_counts;
+
+long print_ca_data(dbCommon *precord, dbFldDes *pdbFldDes, DBLINK* plink, void* pvt) {
+    caLink* pca;
+    ca_counts *counts = (ca_counts*)pvt;
+
+    counts->ncalinks++;
+    pca = (caLink *)plink->value.pv_link.pvt;
```

Comment:
Rather than accessing the private `struct caLink`, external code can use the link support functions.  eg. `dbIsLinkConnected()` and `dbLinkIsVolatile()`.  This is the benefit of also handling pva links.